lib/jekyll/cache.rb

```diff
@@ -12,23 +12,34 @@ class Cache
 
     class << self
       # class-wide cache location
-      attr_accessor :cache_dir
+      attr_reader :cache_dir
 
       # class-wide directive to write cache to disk
       attr_reader :disk_cache_enabled
 
       # class-wide base cache reader
       attr_reader :base_cache
 
+      # Setup cache attributes for given +site+
+      #
+      # site - A Jekyll::Site instance
+      #
+      # Returns nothing
+      def bootstrap(site)
```

Comment:
I'd rather be more explicit about the interface here. Let's change the bootstrap signature a bit:
```suggestion
      def bootstrap(configured_cache_dir, safe_mode)
```

It also removes the dependency on the `site` object, which is really only used here and in `determine_cache_dir`

Reply:
I intentionally added the dependency to `site` because we've no other way to refer to the `site` object. Though, I don't remember why I needed to reference the `site` object anyway.. :thinking:

Reply:
The `determine_cache_dir` is wired to check if the configured cache-dir isn't the same as the current source-dir and adapt accordingly. So, we need the `site` object here as this logic can't be moved into the `Jekyll::Site` class.